experimental/saiexperimentaldashroutinggroup.h

```diff
@@ -0,0 +1,136 @@
+ *
+ * @file    saiexperimentaldashroutinggroup.h
+ *
+ * @brief   This module defines SAI extensions for DASH ROUTING GROUP
+ *
+ * @warning This module is a SAI experimental module
+ */
+
+#if !defined (__SAIEXPERIMENTALDASHROUTINGGROUP_H_)
+#define __SAIEXPERIMENTALDASHROUTINGGROUP_H_
+
+#include <saitypesextensions.h>
+
+/**
+ * @defgroup SAIEXPERIMENTALDASH_ROUTING_GROUP SAI - Experimental: DASH ROUTING GROUP specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief Attribute ID for dash_routing_group_routing_group
+ */
+typedef enum _sai_routing_group_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_ROUTING_GROUP_ATTR_START,
+
+    /**
+     * @brief Action set_routing_group_attr parameter ADMIN_STATE
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_ROUTING_GROUP_ATTR_ADMIN_STATE = SAI_ROUTING_GROUP_ATTR_START,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_ROUTING_GROUP_ATTR_END,
+
+    /** Custom range base value */
+    SAI_ROUTING_GROUP_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_ROUTING_GROUP_ATTR_CUSTOM_RANGE_END,
+
+} sai_routing_group_attr_t;
+
+/**
+ * @brief Create dash_routing_group_routing_group
+ *
+ * @param[out] routing_group_id Entry id
+ * @param[in] switch_id Switch id
+ * @param[in] attr_count Number of attributes
+ * @param[in] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_create_routing_group_fn)(
+        _Out_ sai_object_id_t *routing_group_id,
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list);
+
+/**
+ * @brief Remove dash_routing_group_routing_group
+ *
+ * @param[in] routing_group_id Entry id
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_remove_routing_group_fn)(
+        _In_ sai_object_id_t routing_group_id);
+
+/**
+ * @brief Set attribute for dash_routing_group_routing_group
+ *
+ * @param[in] routing_group_id Entry id
+ * @param[in] attr Attribute
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_set_routing_group_attribute_fn)(
+        _In_ sai_object_id_t routing_group_id,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Get attribute for dash_routing_group_routing_group
+ *
+ * @param[in] routing_group_id Entry id
+ * @param[in] attr_count Number of attributes
+ * @param[inout] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t (*sai_get_routing_group_attribute_fn)(
+        _In_ sai_object_id_t routing_group_id,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list);
+
+typedef struct _sai_dash_routing_group_api_t
```

Comment:
since the routing entry in DASH is outbound_routing_entry, do you mind to change this to outbound_routing_group to make them align?

also, I believe it is better to merged into outbound routing header, we can use the annotation dash_api in the p4 file to specify the API group.

Reply:
Changed and merged